camp/camp-brooklyn/src/test/java/org/apache/brooklyn/camp/brooklyn/ObjectsYamlTest.java

```diff
@@ -275,6 +288,57 @@ public void testBrooklynObjectWithFunction() throws Exception {
         Assert.assertTrue(testObjectObject instanceof SimpleTestPojo, "Expected a SimpleTestPojo: "+testObjectObject);
     }
 
+    @Test
+    public void testBrooklynObjectWithParameterisedConstructor() throws Exception {
```

Comment:
Looks good. Would also be nice to have a test that used a deferred supplier. But given there are no existing tests in `ObjectsYamlTest` that do that, I see why you didn't either!

We can do that in a separate PR.